client/packages/common/src/ui/layout/tables/DataTable.tsx

```diff
@@ -131,13 +132,15 @@ const DataTableComponent = <T extends RecordWithId>({
   const t = useTranslation();
   const { setRows, setDisabledRows, setFocus } = useTableStore();
   const [clickFocusedRow, setClickFocusedRow] = useState(false);
-  const [displayColumns, setDisplayColumns] = useState(columns);
-  const columnsToDisplay = React.useMemo(
-    () =>
-      columns.filter(c => displayColumns.map(({ key }) => key).includes(c.key)),
-    [displayColumns, columns]
+  const { columnDisplayState, toggleColumn } = useColumnDisplayState(
+    id,
+    columns
   );
 
+  const columnsToDisplay = React.useMemo(() => {
+    return columns.filter(c => columnDisplayState[String(c.key)] ?? true);
+  }, [columns, columnDisplayState]);
+
```

Comment:
:nit tidier without the return
```suggestion
  const columnsToDisplay = React.useMemo(
    () => columns.filter(c => columnDisplayState[String(c.key)] ?? true),
    [columns, columnDisplayState]
  );
```